src/sentry/migrations/0020_hide_environment_none_20191126.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+def hide_environent_none(apps, schema_editor):
+    """
+    Hide environments that are named none, since they're blacklisted and no longer can be created.
+
+    We should iterate over each environment row individually in python instead so that we don't lock the DB up. This is
+    far slower but much safer
+    """
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    for env in Environment.objects.all():
+        if env.name == 'none':
+            for project in EnvironmentProject.objects.filter(environment_id=env.id):
+                project.is_hidden = True
+                project.save()
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Adding indexes to large tables. These indexes should be created concurrently,
+    #   unfortunately we can't run migrations outside of a transaction until Django
+    #   1.10. So until then these should be run manually.
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
```

Comment:
This now needs to be True. It’ll be too slow for normal deploy.

Reply:
Just as a rule of thumb, if you're doing a migration that might take a long time you always want to mark this as `is_dangerous`. You'll also need to co-ordinate with #ops after this is landed so that someone can run it for you.

You can try estimate how long a migration might take by just checking out redash and seeing how many rows you're likely to hit based on your queries.